homeassistant/components/roborock/__init__.py

```diff
@@ -141,14 +141,9 @@ async def setup_device(
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Handle removal of an entry."""
-    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-    if unload_ok:
-        await asyncio.gather(
-            *(
-                coordinator.release()
-                for coordinator in hass.data[DOMAIN][entry.entry_id].values()
-            )
-        )
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        for coordinator in hass.data[DOMAIN][entry.entry_id].values():
+            coordinator.release()
```

Comment:
This call was previously awaited, not it isn't. Is that expected?

Reply:
It's definitely fine, it does the same things in a sync manner. But doing async is better, so I am marking this PR as a draft until #107547 is merged

Reply:
Resolved in latest commit